### PR TITLE
Making Changes to Several Tutorials

### PR DESCRIPTION
**Pull request by smithis7**
**Description**

Changes include:
- Some general formatting of notebooks.
- Adding the `download_atom_data` function to every notebook that needs it so that the notebooks will run properly in binder.
  - In doing this, I fixed the print statement in `tardis/io/atom_data/util.py` to use the correct import.
- Addressing errors in all tutorials that had them (except for `docs/io/configuration/components/models/Custom_TARDIS_Model_Tutorial.ipynb` which is supposed to have those error -- I will be returning to this notebook in a later PR).
- Clearing the output in all tutorials (besides `docs/io/configuration/components/models/Custom_TARDIS_Model_Tutorial.ipynb`) so that they will be executed by nbsphinx.
- Adding in a small HDF to `docs/io/visualization/` so that the visualization notebooks can run. Because I cleared the output in `docs/io/visualization/sdec_plot.ipynb`, **this PR will still decrease the overall size of the repo**.
- Removing some of the plotly plots from `docs/io/visualization/sdec_plot.ipynb` so that the built file is <100 MB (GH pages requirement).
- Deleting a duplicate notebook (`docs/io/configuration/components/models/densitycust/Custom_Density_And_Boundary_Velocities.ipynb`).
- Adding a note to the quickstart guide.
- Rewriting `docs/io/optional/callback_example.ipynb`.
- Adding more text to `docs/io/configuration/read_configuration.ipynb` and including information about reading configuration dictionaries.
- Moving `docs/io/output/hdf_writer.ipynb` to be under `docs/outdated`, allowing me to make our section about writing a simulation to an HDF much more simple.

_NOTE: the docs building now takes way too long. I am exploring options to fix this._

**Motivation and context**

Now, nearly all of the TARDIS tutorials we want to work are actually working and formatted properly.

**How has this been tested?**
- [ ] Testing pipeline.
- [x] Docs built locally and on GitHub. 

**Examples**

https://smithis7.github.io/tardis/branch/notebook_format_doc/

**Type of change**

- [x] Bug fix. 
- [ ] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [x] I have updated the documentation accordingly.
    - [x] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: docs/quickstart/quickstart.ipynb [hunk 1]
```diff
@@ -43,7 +43,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# the data is automatically downloaded\n",
+    "# The data is automatically downloaded\n",
```
Comment (andrewfullard) [inline docs/quickstart/quickstart.ipynb:46]: The data *are*. Data is plural

File: docs/outdated/hdf_writer.ipynb [hunk 2]
```diff
@@ -269,7 +269,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.8"
+   "version": "3.7.10"
```
Comment (andrewfullard) [inline docs/outdated/hdf_writer.ipynb:272]: Does this match the python version in the tardis environment?

File: docs/io/configuration/read_configuration.ipynb [hunk 3]
```diff
@@ -4,12 +4,14 @@
    "cell_type": "markdown",
```
Comment (andrewfullard) [inline docs/io/configuration/read_configuration.ipynb:4]: <p>Is this a sphinx note? Also typo recommend.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1763/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Comment by andrewfullard** [comment]
I think we definitely need to be careful about removing information (e.g. the SDEC plot examples) but otherwise generally good

**Review by andrewfullard — APPROVED** [review]

File: docs/io/visualization/generating_widgets.ipynb [hunk 4]
```diff
@@ -14,384 +14,28 @@
    "cell_type": "markdown",
```
Comment (jaladh-singhal) [inline docs/io/visualization/generating_widgets.ipynb:14]: <p>Nice note at bottom :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1763/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

File: docs/quickstart/quickstart.ipynb [hunk 5]
```diff
@@ -2,14 +2,13 @@
  "cells": [
```
Comment (jaladh-singhal) [inline docs/quickstart/quickstart.ipynb:2]: <p>I think @wkerzendorf mentioned to show INFO level logs!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1763/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

**Review by jaladh-singhal — COMMENTED** [review]
These changes are so thoughtful - awesome work @smithis7 :tada: 

There are few things:
- Can you rename "TARDIS Grid Tutorial" to "TARDIS Grid" - I think tutorial looks bit odd considering how its sibling sections are named
- It needs rebase, there are merge conflicts in logging notebook
- Your created hdf don't show up Ar for real packet population but shows up for virtual - can you please fix that we really want to show same colorscale in both packets_mode to avoid confusion
- About adding hdf to visualization folder seems reasonable to make it run on binder and since you've freed lot of space by removing outputs. But I'd ask @wkerzendorf to have a final say on this!

File: docs/quickstart/quickstart.ipynb (hunk 5 above)
Comment (smithis7) [reply docs/quickstart/quickstart.ipynb:2]: My thinking is that we want to show the defaults in quickstart. @wkerzendorf to my memory talked about changing it so "info" is the default.

**Comment by smithis7** [comment]
> These changes are so thoughtful - awesome work @smithis7 🎉
> 
> There are few things:
> 
> * Can you rename "TARDIS Grid Tutorial" to "TARDIS Grid" - I think tutorial looks bit odd considering how its sibling sections are named
> * It needs rebase, there are merge conflicts in logging notebook
> * Your created hdf don't show up Ar for real packet population but shows up for virtual - can you please fix that we really want to show same colorscale in both packets_mode to avoid confusion
> * About adding hdf to visualization folder seems reasonable to make it run on binder and since you've freed lot of space by removing outputs. But I'd ask @wkerzendorf to have a final say on this!

I have rebased and fixed the hdf issue. And I did talk to Wolfgang about adding the HDF before I submitted this PR.

For the "TARDIS Grid" page, there are a few things on that that are being talked about (i.e. the filename will be changed and we may move the file and possibly come up with a different name), so I think we should make any changes to that page all at once. And, Marc will make the changes to the file name so it still shows him as the page author.

**Review by jaladh-singhal — APPROVED** [review]
Sorry, it took me so long to review this - well done!

**Pull request merged**